src/ToolBox/SOS/lldbplugin/CMakeLists.txt

```diff
@@ -18,6 +18,10 @@ add_definitions(-DPAL_STDCPP_COMPAT)
 set(ENABLE_LLDBPLUGIN ${CLR_CMAKE_PLATFORM_UNIX} CACHE BOOL "Enable building the SOS plugin for LLDB.")
 set(REQUIRE_LLDBPLUGIN ${CLR_CMAKE_PLATFORM_LINUX} CACHE BOOL "Require building the SOS plugin for LLDB.")
 
+if(CLR_CMAKE_PLATFORM_FREEBSD)
+set(REQUIRE_LLDBPLUGIN true)
+endif (CLR_CMAKE_PLATFORM_FREEBSD)
```

Comment:
This is setting the non-cache variable, but Linux set the cache variable.  Should this be

```cmake
set(REQUIRE_LLDBPLUGIN true CACHE BOOL "Require building the SOS plugin for LLDB." FORCE)
```
?

Or maybe restructuring the existing set(REQUIRE_LLDBPLUGIN  ... CACHE) to an if/else itself.

Reply:
This is what we do for SKIP_LLDBPLUGIN and ARM64. I did try to update the first set but cmake does not seems to like expression there with OR. maybe @janvorli would know.

Reply:
Theoretically we could do the find_path changes now, and then figure out what to for making this move from ENABLE-only to REQUIRE in a followup, ja?

Reply:
What this change really does is that it sets the REQUIRE_LLDBPLUGIN without giving the user the ability to override it from the command line using the `-DREQUIRE_LLDBPLUGIN`. The cache variables are basically used as default values that are used unless they are overridden from the command line or by a `set` without `CACHE`. It seems that to preserve the behavior that we have for Linux, we should just update the value in the set above. @wfurt  you can just do it this way:
```cmake
if (${CLR_CMAKE_PLATFORM_LINUX} OR ${CLR_CMAKE_PLATFORM_FREEBSD})
 set(PLATFORM_SUPPORTS_LLDB_PLUGIN true)
endif()
```
and then use the `PLATFORM_SUPPORTS_LLDB_PLUGIN` instead of the `CLR_CMAKE_PLATFORM_LINUX` in the `set` commands with `CACHE`.

Reply:
ok. I was thinking bout that as well but I was not sure about I was not sure about all implications for new OSes. I like the proposal and I'll update PR accordingly.